src/main/kotlin/io/getunleash/metrics/MetricsReporter.kt

```diff
@@ -84,11 +83,15 @@ class HttpMetricsReporter(
     private var bucket: Bucket = Bucket(start = started)
 
     override fun reportMetrics() {
+        val stop = Date()
+        val bucketRef = bucket
+        bucket = Bucket(start = stop)
+        val clonedMetrics = bucketRef.copy(stop = stop)
```

Comment:
Immediately swap variables so new counts are against the new bucket, only then make a copy. If due to a race condition a count is still counted against the old bucket it'll still be counted, so we should not lose data